generators/openapi-client/files.js

```diff
@@ -177,23 +155,36 @@ function writeFiles() {
                 return;
             }
 
+            this.javaDir = `${constants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
             const mainClassFile = `${this.javaDir + this.getMainClassName()}.java`;
 
             this.rewriteFile(
                 mainClassFile,
                 'import org.springframework.core.env.Environment;',
-                'import org.springframework.context.annotation.ComponentScan;'
+                'import org.springframework.context.annotation.ComponentScan;\n' +
+                    'import org.springframework.context.annotation.FilterType;'
             );
 
             const componentScan =
-                `${'@ComponentScan( excludeFilters = {\n    @ComponentScan.Filter('}${this.packageName}` +
-                '.client.ExcludeFromComponentScan.class)\n})';
+                '@ComponentScan( excludeFilters = {\n' +
+                '   @ComponentScan.Filter(type = FilterType.REGEX, ' +
+                `pattern = "${this.packageName}.client.*.ClientConfiguration")\n})`;
             this.rewriteFile(mainClassFile, '@SpringBootApplication', componentScan);
-
-            this.template(
-                'src/main/java/package/client/_ExcludeFromComponentScan.java',
-                `${this.javaDir}/client/ExcludeFromComponentScan.java`
-            );
         }
     };
 }
+
+function removeClientFiles(cliName) {
+    const baseCliPackage = `${this.packageName}.client.`;
+    const cliPackage = `${baseCliPackage}${_.toLower(cliName)}`;
+    const snakeCaseCliPackage = `${baseCliPackage}${_.snakeCase(cliName)}`;
+    removeFolder(path.resolve(constants.SERVER_MAIN_SRC_DIR, ...cliPackage.split('.')));
```

Comment:
If bonded with the generator can be this.removeFolder.
https://github.com/jhipster/generator-jhipster/blob/69b7aac3e123a8f6cc8680d79b3f38091f46ec50/generators/generator-base-private.js#L444